learningequality/kolibri
# Fix: Side panel closing confirmation & logic in child component

## Summary


The issue here was that the logic for whether or not the panel should be closed or whether or not we need to show a confirmation modal was initially built into the SectionSidePanel component. This component is a wrapper component and now that the workingResourcePool lives solely within the ResourceSelection component, there is no access to it in the parent.

The solution was to pass a prop into each child component which will need to be handled in all of the child components (section edit, question replacement). This prop tells the child that we're trying to close the side panel.

This allows each child component then to define its own logic and messaging around confirming the panels closure with the user. This results in the child component also needing to emit two events which tell the SectionSidePanel if the panel should actually be closed, resulting in the user being returned to the root quiz creation route.


## Reviewer guidance


#### Testing

- Proceed to add questions to a section
- Change your selection (click one resource's checkbox)
- Try to close the side panel by the X or by clicking outside of the side panel
- See the modal window pops up
- When "Cancel" is clicked, you still see the side panel and if you try to close again, you get the confirmation modal again
- When "Continue" is clicked, you see the side panel closed
- Try this again on the same section by clicking "Options" -> "Edit section" -> "Select resources..." and then make a change to your selection and the same confirmation modal flow as noted above should be done.

- **The same workflow should present itself when in the SectionEditor**
